src/extension.ts

```diff
@@ -140,304 +57,130 @@ namespace ActionableNotification {
-          }
+        actionableNotificationSupport: true,
+        openSettingsCommandSupport: true
+      }
+    },
+    synchronize: {
+      //preferences starting with these will trigger didChangeConfiguration
+      configurationSection: ['xml', '[xml]', 'files.trimFinalNewlines', 'files.trimTrailingWhitespace', 'files.insertFinalNewline']
+    },
+    middleware: {
+      workspace: {
+        didChangeConfiguration: () => {
+          languageClient.sendNotification(DidChangeConfigurationNotification.type, { settings: getXMLSettings(requirementsData.java_home, logfile, externalXmlSettings) });
+          onConfigurationChange();
         }
       }
+    }
+  }
 
-      prepareExecutable(requirements, collectXmlJavaExtensions(extensions.all, getXMLConfiguration().get("extension.jars", [])), context)
-        .then((serverOptions: Executable) => {
-          languageClient = new LanguageClient('xml', 'XML Support', serverOptions, clientOptions);
-          let toDispose = context.subscriptions;
-          let disposable = languageClient.start();
-          toDispose.push(disposable);
-
-          languages.setLanguageConfiguration('xml', getIndentationRules());
-          languages.setLanguageConfiguration('xsl', getIndentationRules());
-
-          return languageClient.onReady().then(() => {
-            //Detect JDK configuration changes
-            disposable = subscribeJDKChangeConfiguration();
-            toDispose.push(disposable);
-
-            // Code Lens actions
-            context.subscriptions.push(commands.registerCommand(Commands.SHOW_REFERENCES, (uriString: string, position: Position) => {
-              const uri = Uri.parse(uriString);
-              workspace.openTextDocument(uri).then(document => {
-                // Consume references service from the XML Language Server
-                let param = languageClient.code2ProtocolConverter.asTextDocumentPositionParams(document, position);
-                languageClient.sendRequest(ReferencesRequest.type, param).then(locations => {
-                  commands.executeCommand(Commands.EDITOR_SHOW_REFERENCES, uri, languageClient.protocol2CodeConverter.asPosition(position), locations.map(languageClient.protocol2CodeConverter.asLocation));
-                })
-              })
-            }));
-
-            setupActionableNotificationListener(languageClient);
+  const serverOptions: Executable = await prepareExecutable(
+    requirementsData, collectXmlJavaExtensions(extensions.all, getXMLConfiguration().get("extension.jars", [])), context);
 
-            // Handler for 'xml/executeClientCommand` request message that executes a command on the client
-            languageClient.onRequest(ExecuteClientCommandRequest.type, async (params: ExecuteCommandParams) => {
-              return await commands.executeCommand(params.command, ...params.arguments);
-            });
+  languageClient = new LanguageClient('xml', 'XML Support', serverOptions, clientOptions);
```

Comment:
I will move all LSP client code into the `client/xmlClient.ts` like HTML https://github.com/microsoft/vscode/blob/master/extensions/html-language-features/client/src/htmlClient.ts and I will move tagClosing.ts in this `client` folder.

Reply:
IMHII think it should be nice to have the following folders:

 * `client` 
 * `server`

to be consistent with HTML language server for instance https://github.com/microsoft/vscode/tree/main/extensions/html-language-features